.github/workflows/pub-publish.yml

```
@@ -0,0 +1,36 @@
+name: Publish to pub.dev
+
+## Caller of this workflow should use it as follows:
+## jobs:
+##   publish:
+##     uses: dart-lang/setup-dart/.github/workflows/pub-publish.yml
+##     # with:
+##     #   working-directory: path/to/sub/directory
+
+on:
+  workflow_call:
+    inputs:
+      working-directory:
+        required: false
+        type: string
+
+jobs:
+  pub-publish:
+    permissions:
```

```suggestion
  publish:
    name: 'Publish to pub.dev'
    permissions:
```

nit: maybe we can give it a name :D